plugins/by-name/oil-git-status/default.nix

```
@@ -0,0 +1,66 @@
+{
+  lib,
+  config,
+  ...
+}:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "oil-git-status";
+  packPathName = "oil-git-status.nvim";
+  package = "oil-git-status-nvim";
+
+  description = ''
+    Add Git Status to oil.nvim directory listings.
+
+    Git status is added to the listing asynchronously after creating the oil directory
+
+    listing so it won't slow oil down on big repositories. The plugin puts the status in two new sign columns
+
+    the left being the status of the index, the right being the status of the working directory
+
+    `win_options.signcolumn` needs to be configured in `plugins.oil.settings`:
+    ```nix
+    plugins.oil = {
+      enable = true;
+      settings = {
+        win_options = {
+          signcolumn = "yes:2,";
+        };
+      };
+    };
+    ```
+  '';
+
+  maintainers = [ lib.maintainers.FKouhai ];
+
+  # Ensure oil-git-status is set up after oil is loaded
+  configLocation = lib.mkOrder 1100 "extraConfigLua";
+
+  dependencies = [
+    "git"
+  ];
+
+  extraConfig = cfg: {
+    warnings = lib.nixvim.mkWarnings "plugins.oil-git-status" [
+      {
+        when = !config.plugins.oil.enable;
+        message = ''
+          This plugin requires `plugins.oil` to be enabled
+          plugins.oil.enable = true;
```

This is kinda subjective, but if we interpolate the _actual_ option, users will see either `plugins.oil.enable` or `programs.nixvim.plugins.oil.enable` depending on whether their nixvim configuration is a `programs.nixvim` submodule.

This also has the advantage of ensuring no typos.

```suggestion
              ${options.plugins.oil.enable} = true;
```

To do this, you need the `options` module arg of course.

Also, nit: I would indent the code example and add a colon to the end of the previous line, to be consistent with the other warning.

---

yep I like that :)